datafusion/physical-plan/src/topk/mod.rs

```diff
@@ -232,6 +248,105 @@ impl TopK {
         // which means the top K won't change and the computation can be finished early.
         self.attempt_early_completion(&batch)?;
 
+        if updated {
+            // update the filter representation of our TopK heap
+            self.update_filter()?;
+        }
+
+        Ok(())
+    }
+
+    /// Update the filter representation of our TopK heap
```

Comment:
FWIW I recommend adding an example to this doc string showing what the filter expression would be for a multi-column sort order